opensearch/src/main/java/org/opensearch/sql/opensearch/request/AggregateAnalyzer.java

```diff
@@ -266,4 +274,26 @@ private static CompositeValuesSourceBuilder<?> createTermsSourceBuilder(
 
     return sourceBuilder;
   }
+
+  private static ValuesSourceAggregationBuilder<?> createBucketAggregation(
+      Integer group, FieldExpressionCreator fieldExpressionCreator) {
+    NamedFieldExpression groupExpr = fieldExpressionCreator.create(group);
+    // TODO: support histogram bucket(i.e. PPL span expression)
+    // https://github.com/opensearch-project/sql/issues/3384
+    return createTermsAggregationBuilder(groupExpr);
+  }
+
+  private static TermsAggregationBuilder createTermsAggregationBuilder(
+      NamedFieldExpression groupExpr) {
+
+    TermsAggregationBuilder sourceBuilder =
+        new TermsAggregationBuilder(groupExpr.getRootName()).size(AGGREGATION_BUCKET_SIZE);
```

Comment:
what if buckets more than AGGREGATION_BUCKET_SIZE?

Reply:
Only 1000 buckets return. The default size is 10. This setting is aligned with previous implementation of composite agg which set 1000 buckets.